LanguageExt.Core/Extensions/ObjectExt.cs

```diff
@@ -45,22 +45,28 @@ public static class ObjectExt
 
         private static class Check<T>
         {
+            static readonly T DefaultT;
             static readonly bool IsReferenceType;
             static readonly bool IsNullable;
             static readonly EqualityComparer<T> DefaultEqualityComparer;
 
             static Check()
             {
+                DefaultT = default(T);
                 IsNullable = Nullable.GetUnderlyingType(typeof(T)) != null;
                 IsReferenceType = !typeof(T).GetTypeInfo().IsValueType;
                 DefaultEqualityComparer = EqualityComparer<T>.Default;
             }
 
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal static bool IsDefault(T value) =>
+                DefaultEqualityComparer.Equals(value, DefaultT);
+
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             internal static bool IsNull(T value) =>
                 IsNullable
-                    ? value.Equals(default(T))
-                    : IsReferenceType && DefaultEqualityComparer.Equals(value, default(T));
+                    ? value.Equals(DefaultT)
```

Comment:
This should stay as `default(T)`, it's more efficient than accessing the `DefaultT` field.